src/vanilla/store.ts

```diff
@@ -288,13 +288,14 @@ const buildStore = (getAtomState: StoreArgs[0]): Store => {
     pending: Pending | undefined,
     atom: Atom<Value>,
     atomState: AtomState<Value>,
-    force?: (a: AnyAtom) => boolean,
+    dirtyAtoms?: Set<AnyAtom>,
+    force?: boolean,
   ): AtomState<Value> => {
     // See if we can skip recomputing this atom.
-    if (!force?.(atom) && isAtomStateInitialized(atomState)) {
+    if (!force && isAtomStateInitialized(atomState)) {
```

Comment:
@dai-shi I'm not sure why but removing `!force` from this makes all tests pass... do you know why this would work?

Reply:
cd67e40 --- yeah. I think we miss some optimization tests. I'll try to add one.

Reply:
Hmm, it actually seems fine, because in normal cases (no reordering), dirtyAtoms doesn't include dependencies. So, no performance drawback.